### DISCUSSION TITLE
unable to upgrade from old v. 3.3.6 due to duplicates

### DISCUSSION THREAD

**Discussion by jancovic** [netbox-community/netbox#13134] [General]
Hi

I am unable to upgrade netbox from 3.3.6  to latest ver. due to this braking changes in 3.4.0
...Device and virtual machine names are no longer case-sensitive. Attempting to create e.g. "device1" and "DEVICE1" within the same site will raise a validation error

Why this major change was implemented ?
I believe I am not only one who has problem with upgrade.
Any idea how to fix this effective way ?
When I remove/rename/fix  one device and run upgrade, then another duplicate is found...there can be hundreds...:(
How can I make list of all "duplicates" from DB in once ?
 How you fixed this ?
 

```
django.db.utils.IntegrityError: could not create unique index "dcim_device_uniqu
DETAIL:  Key (lower(name::text), site_id)=(underl-0014, 1) is duplicated.
```

**Comment by candlerb** [comment]
> Why this major change was implemented ?

I can't speak for the authors, but I would guess because it was an oversight to allow it in the first place: e.g. allowing a device called "foo" and another device called "Foo" in the same site for the same tenant, is confusing and error-prone.  In that case, you might as well allow two devices called "foo".

> Any idea how to fix this effective way ?
> When I remove/rename/fix one device and run upgrade, then another duplicate is found...there can be hundreds...:(

You can use a SQL query to find them:

```
sudo -u postgres psql netbox
...
select lower(name) as lname,site_id,count(*) from dcim_device group by lower(name),site_id having count(*) > 1;
```

How to fix them depends on what your naming policy is. For example, given two devices called "UNDERL-0014" and "underl-0014" in the same site, what are your preferred new names?

If you want to script the fixups, you should do it via [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/), a custom script, or the REST API.  Direct SQL updates to `dcim_device.name` are not a good idea, since there's a derived column `_name` which also needs to be updated - it's derived via Django ORM code, it's not a postgres generated column.

**Comment by jancovic** [comment]
Thank you very much ! I had just this one duplicate. After changing name, upgrade finished OK

**Reply by muhammadhusain078** [reply]
How to change the name of the duplicate one, sorry i have no idea with postgres

**Reply by muhammadhusain078** [reply]
i have the same issue

**Reply by candlerb** [reply]
Easiest to change it in Netbox via the web UI. This may mean restoring your system to how it was prior to the attempted upgrade.
